src/common/LoginConfig.ts

```diff
@@ -2,85 +2,111 @@ import * as core from '@actions/core';
 import { FormatType, SecretParser } from 'actions-secret-parser';
 
 export class LoginConfig {
+    static readonly AUTH_TYPE_SERVICE_PRINCIPAL = "SERVICE_PRINCIPAL";
+    static readonly AUTH_TYPE_IDENTITY = "IDENTITY";
     static readonly azureSupportedCloudName = new Set([
         "azureusgovernment",
         "azurechinacloud",
         "azuregermancloud",
         "azurecloud",
         "azurestack"]);
 
+    static readonly azureSupportedAuthType = new Set([
+        LoginConfig.AUTH_TYPE_SERVICE_PRINCIPAL,
+        LoginConfig.AUTH_TYPE_IDENTITY]);
+
+    authType: string;
     servicePrincipalId: string;
     servicePrincipalKey: string;
     tenantId: string;
     subscriptionId: string;
     resourceManagerEndpointUrl: string;
     allowNoSubscriptionsLogin: boolean;
-    enableOIDC: boolean;
     environment: string;
     enableAzPSSession: boolean;
     audience: string;
     federatedToken: string;
 
-    constructor() {
-        this.enableOIDC = true;
-    }
-
     async initialize() {
         this.environment = core.getInput("environment").toLowerCase();
         this.enableAzPSSession = core.getInput('enable-AzPSSession').toLowerCase() === "true";
         this.allowNoSubscriptionsLogin = core.getInput('allow-no-subscriptions').toLowerCase() === "true";
+        this.authType = core.getInput('auth-type').toUpperCase();
 
         this.servicePrincipalId = core.getInput('client-id', { required: false });
         this.servicePrincipalKey = null;
         this.tenantId = core.getInput('tenant-id', { required: false });
         this.subscriptionId = core.getInput('subscription-id', { required: false });
 
+        this.readParametersFromCreds();
+
         this.audience = core.getInput('audience', { required: false });
         this.federatedToken = null;
+
+        this.mask(this.servicePrincipalId);
+        this.mask(this.servicePrincipalKey);

```

Comment:
Does servicePrincipalKey actually mean servicePrincipal secret?

Reply:
yes. Renamed `servicePrincipalKey` to `servicePrincipalSecret`.